### PR TITLE
Updating LESS compiler to 0.5.0

### PR DESCRIPTION
**Pull request by aristath**
Latest stable version from https://github.com/leafo/lessphp/releases/tag/v0.5.0

### CODE REVIEW

**Comment by samhotchkiss** [comment]
Can we get some input from tdiv and @jeffgolenski on this?

**Comment by zinigor** [comment]
@aristath Thank you for the PR! Looking at the code I can see that it touches a lot more than just the version of the LESS compiler. Can you please split this change into several PRs that will each handle their own purpose and explain the changes in each of them?

**Comment by georgestephanis** [comment]
@zinigor It will update us to a current version of an external library.  It's a good thing to do, but we need to at least skim the changes.

File: modules/custom-css/custom-css/preprocessors/lessc.inc.php [hunk 1]
```diff
@@ -537,7 +570,7 @@ protected function patternMatch($block, $orderedArgs, $keywordArgs) {
 			return true; // not having enough is handled above
 		} else {
 			$numMatched = $i + 1;
-			// greater than because default values always match
+			// greater than becuase default values always match
```
Comment (georgestephanis) [inline modules/custom-css/custom-css/preprocessors/lessc.inc.php]: Why did this comment change from correctly spelled to misspelled?

File: modules/custom-css/custom-css/preprocessors/lessc.inc.php [hunk 2]
```diff
@@ -1168,36 +1239,56 @@ protected function lib_mix($args) {
 	}
 
 	protected function lib_contrast($args) {
-		if ($args[0] != 'list' || count($args[2]) < 3) {
-			return array(array('color', 0, 0, 0), 0);
-		}
+	    $darkColor  = array('color', 0, 0, 0);
```
Comment (georgestephanis) [inline modules/custom-css/custom-css/preprocessors/lessc.inc.php]: The changes here and on the following lines now have mixed spaces and tabs.  Boo.

**Comment by georgestephanis** [comment]
All the changes look reasonable -- there's some minor whitespace and newly introduced typos, but nothing blocking.  cc: @leafo ^^

**Comment by georgestephanis** [comment]
Crap!  Didn't mean to close the pull request!

File: modules/custom-css/custom-css/preprocessors/lessc.inc.php (hunk 1 above)
Comment (georgestephanis) [inline modules/custom-css/custom-css/preprocessors/lessc.inc.php]: Ah, looks like someone changed it downstream here in Jetpack.  Boo downstream changes.

**Comment by aristath** [comment]
Well, I could fix any indentation issues and re-submit if you want. :)

File: modules/custom-css/custom-css/preprocessors/lessc.inc.php (hunk 1 above)
Comment (kraftbj) [inline modules/custom-css/custom-css/preprocessors/lessc.inc.php]: Sent a PR upstream to https://github.com/leafo/lessphp/pull/601 :)

**Comment by kraftbj** [comment]
Resubmitted PR via an automattic/jetpack branch in #2205

**Pull request closed without merging**